### PR TITLE
Introduce config params for scalable pool

### PR DESCRIPTION
**Pull request by vinser52**
### Description
Introduce config params for scalable pool

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly
- [x] New tests added, especially if they will fail without my changes

### CODE REVIEW

**Comment by vinser52** [comment]
Ref. #844

File: test/pools/scalable_pool.cpp [hunk 1]
```diff
@@ -7,10 +7,148 @@
...
+    umf_result_t ret = umfScalablePoolParamsCreate(nullptr);
+    ASSERT_EQ(ret, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    ret = umfScalablePoolParamsSetGranularity(nullptr, 2 * 1024 * 1024);
+    ASSERT_EQ(ret, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    ret = umfScalablePoolParamsSetGranularity(params, 0);
+    ASSERT_EQ(ret, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    ret = umfScalablePoolParamsSetKeepAllMemory(NULL, true);
```
Comment (kswiecicki) [inline test/pools/scalable_pool.cpp:140]: nit: You alternate between `NULL` and `nullptr`.

File: src/pool/pool_scalable.c [hunk 2]
```diff
@@ -167,19 +173,88 @@ static void tbb_raw_free_wrapper(intptr_t pool_id, void *ptr, size_t bytes) {
     }
 }
 
+umf_result_t
+umfScalablePoolParamsCreate(umf_scalable_pool_params_handle_t *params) {
+    if (!params) {
+        LOG_ERR("params is NULL");
```
Comment (kswiecicki) [inline src/pool/pool_scalable.c:179]: nit: This message could be more expressive, like the memory allocation error log below, `invalid scalable pool params handle` or something.

**Review by kswiecicki — APPROVED** [review]

File: test/pools/scalable_pool.cpp (hunk 1 above)
Comment (vinser52) [reply test/pools/scalable_pool.cpp]: Done

File: src/pool/pool_scalable.c (hunk 2 above)
Comment (vinser52) [reply src/pool/pool_scalable.c]: Done

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**